### ISSUE TITLE
Konnectors should export a Spider instead of running code directly

### ISSUE THREAD

**Issue by ptbrowne** [cozy/cozy-konnector-libs#98]
Greg has shown us a runner that expects a common interface, that would orchestrate the run of a konnector. This decouples the instrumentation (LOGIN_OK, LOGIN_FAILED, SYNC_SUCCESSFUL) of the konnector from its implementation.

```js
import { run } from 'cozy-konnector-libs'

run({
  login: function () {
    return request(....)
  },

  sync: function () {
    return request(....).then(addData)
  }
})
```

The `run` function then can instrument the konnector without changing its implementation.

```js
const run = async konnector => {
  const ok = await konnector.login()
  if (ok) {
    await realtime('LOGIN_OK')
  } else {
    await realtime('LOGIN_FAILED')
    process.exit(1)
  }
  await realtime('START_SYNC')
  await konnector.sync()
  await realtime('SYNC_SUCCESSFUL')
}
```

**Comment by gregorylegarec** [comment]
Thanks for opening this issue @ptbrowne. What I had in mind was to use an [inversion of control](https://en.wikipedia.org/wiki/Inversion_of_control) pattern. The idea is to let `cozy-konnector-libs` run the konnectors in a way or another.

This could improve
* konnector development (less code for contributor, do dependencies and no instanciation)
* testing and mocking (`cozy-konnector-libs` could provide a set of tests to validate a konnector signature, or test directly a call by mocking internal method for fetching for example)

My first POC is on this [branch on the débug konnector](https://github.com/gregorylegarec/cozy-konnector-debug/blob/poc/runner/index.js). I also started a full working example for Trainline but the code is not in the computer I am working with. I will share this example in one week.

**Comment by ptbrowne** [comment]
Yes I am OK with the inversion of control. This is what is shown with the `run` function. At the end, it would not be the user that would directly call it. The user of cozy-konnector-libs would only export the interface. We would at first expose the run function to have a smooth transition path.

**Comment by doubleface** [comment]
I am also OK with the inversion of control but it is important to make the migration path the easiest.
